### PR TITLE
Move service interfaces away from implementations

### PR DESCRIPTION
**Pull request by raffclar**

### CODE REVIEW

File: src/ECS/Registry.cpp [hunk 1]
```diff
@@ -9,8 +9,8 @@
 
 #include "Registry.h"
 
+#include "ECS/Systems/Implementations/RenderingSystem.h"
```
Comment (bwrsandman) [inline src/ECS/Registry.cpp:12]: Already in ECS and why is registry not using an interface?

File: src/ECS/Systems/Implementations/DynamicsSystem.cpp [hunk 2]
```diff
@@ -11,21 +11,20 @@
...
-#include <glm/gtx/rotate_vector.hpp>
-
 #include "3D/LandBlock.h"
 #include "3D/LandIsland.h"
+#include "BulletCollision/BroadphaseCollision/btDbvtBroadphase.h"
+#include "BulletCollision/CollisionDispatch/btCollisionDispatcher.h"
+#include "BulletCollision/CollisionDispatch/btCollisionObject.h"
+#include "BulletCollision/CollisionDispatch/btDefaultCollisionConfiguration.h"
+#include "BulletDynamics/ConstraintSolver/btSequentialImpulseConstraintSolver.h"
+#include "BulletDynamics/Dynamics/btDiscreteDynamicsWorld.h"
```
Comment (bwrsandman) [inline src/ECS/Systems/Implementations/DynamicsSystem.cpp:21]: Bullet is a System-level include, they use `<`, `>`

File: src/ECS/Systems/Implementations/RenderingSystem.cpp [hunk 3]
```diff
@@ -21,6 +19,7 @@
 #include "Graphics/ShaderManager.h"
 #include "Locator.h"
 #include "Resources/Resources.h"
+#include "glm/gtx/transform.hpp"
```
Comment (bwrsandman) [inline src/ECS/Systems/Implementations/RenderingSystem.cpp:22]: Use angle brackets for system level includes

File: src/ECS/Systems/Implementations/RenderingSystem.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include "3D/AllMeshes.h"
+#include "ECS/Systems/Interfaces/RenderingSystemInterface.h"
+#include "bgfx/bgfx.h"
+#include "entt/entt.hpp"
+#include "glm/mat4x4.hpp"
```
Comment (bwrsandman) [inline src/ECS/Systems/Implementations/RenderingSystem.h:19]: Use angle brackets for system level includes

File: src/ECS/Systems/Interfaces/RenderingSystemInterface.h [hunk 5]
```diff
@@ -1,27 +1,21 @@
...
-
-#include <bgfx/bgfx.h>
-#include <entt/entt.hpp>
-#include <glm/mat4x4.hpp>
-
-#include "3D/AllMeshes.h"
 #include "Graphics/Mesh.h"
+#include "bgfx/bgfx.h"
+#include "entt/entt.hpp"
+#include "glm/mat4x4.hpp"
```
Comment (bwrsandman) [inline src/ECS/Systems/Interfaces/RenderingSystemInterface.h:16]: Use angle brackets for system level includes

File: src/ECS/Systems/Interfaces/TownSystemInterface.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ ******************************************************************************/
+
+#pragma once
+
+#include "entt/fwd.hpp"
+#include "glm/fwd.hpp"
```
Comment (bwrsandman) [inline src/ECS/Systems/Interfaces/TownSystemInterface.h:13]: Use angle brackets for system level includes

**Review by bwrsandman — CHANGES_REQUESTED** [review]
I get moving implementations to a separate directory but why move the interfaces?
They should be as easy access as possible.

File: src/ECS/Registry.cpp (hunk 1 above)
Comment (raffclar) [reply src/ECS/Registry.cpp]: > I get moving implementations to a separate directory but why move the interfaces? They should be as easy access as possible.

I moved the interfaces to ECS/Systems

Comment (raffclar) [reply src/ECS/Registry.cpp]: > Already in ECS and why is registry not using an interface?

It should be. In fact, it is indirectly here.

Comment (raffclar) [reply src/ECS/Registry.cpp]: Made it use the interface directly

File: src/3D/Camera.cpp [hunk 7]
```diff
@@ -18,7 +18,7 @@
 
 #include "3D/LandIsland.h"
 #include "ECS/Registry.h"
-#include "ECS/Systems/DynamicsSystem.h"
+#include "ECS/Systems/Implementations/DynamicsSystem.h"
```
Comment (bwrsandman) [inline src/3D/Camera.cpp:21]: Why not the interface?

File: src/Debug/Console.cpp [hunk 8]
```diff
@@ -17,7 +17,7 @@
 #include "3D/Camera.h"
 #include "ECS/Components/Transform.h"
 #include "ECS/Registry.h"
-#include "ECS/Systems/DynamicsSystem.h"
+#include "ECS/Systems/Implementations/DynamicsSystem.h"
```
Comment (bwrsandman) [inline src/Debug/Console.cpp:20]: Why not the interface?

File: src/Debug/Gui.cpp [hunk 9]
```diff
@@ -39,13 +39,13 @@
...
 #include <ECS/Components/Villager.h>
 #include <ECS/Registry.h>
-#include <ECS/Systems/LivingActionSystem.h>
 #include <Game.h>
 #include <GameWindow.h>
 #include <Locator.h>
 #include <Resources/Resources.h>
 
 #include "Console.h"
+#include "ECS/Systems/Implementations/LivingActionSystem.h"
```
Comment (bwrsandman) [inline src/Debug/Gui.cpp:48]: Why not the interface?

File: src/ECS/Archetypes/AbodeArchetype.cpp [hunk 10]
```diff
@@ -18,7 +18,7 @@
 #include "ECS/Components/Mesh.h"
 #include "ECS/Components/Transform.h"
 #include "ECS/Registry.h"
-#include "ECS/Systems/TownSystem.h"
+#include "ECS/Systems/Implementations/TownSystem.h"
```
Comment (bwrsandman) [inline src/ECS/Archetypes/AbodeArchetype.cpp:21]: Why not the interface?

File: src/ECS/Archetypes/VillagerArchetype.cpp [hunk 11]
```diff
@@ -20,7 +20,7 @@
 #include "ECS/Components/Villager.h"
 #include "ECS/Components/WallHug.h"
 #include "ECS/Registry.h"
-#include "ECS/Systems/TownSystem.h"
+#include "ECS/Systems/Implementations/TownSystem.h"
```
Comment (bwrsandman) [inline src/ECS/Archetypes/VillagerArchetype.cpp:23]: Why not the interface?

File: src/ECS/Registry.cpp [hunk 12]
```diff
@@ -9,8 +9,8 @@
 
 #include "Registry.h"
 
+#include "ECS/Systems/RenderingSystemInterface.h"
```
Comment (bwrsandman) [inline src/ECS/Registry.cpp:12]: Already in `ECS/`

File: src/Game.cpp [hunk 13]
```diff
@@ -41,12 +41,12 @@
...
-#include "ECS/Systems/LivingActionSystem.h"
-#include "ECS/Systems/PathfindingSystem.h"
-#include "ECS/Systems/RenderingSystem.h"
-#include "ECS/Systems/TownSystem.h"
+#include "ECS/Systems/Implementations/CameraBookmarkSystem.h"
+#include "ECS/Systems/Implementations/DynamicsSystem.h"
+#include "ECS/Systems/Implementations/LivingActionSystem.h"
+#include "ECS/Systems/Implementations/PathfindingSystem.h"
+#include "ECS/Systems/Implementations/RenderingSystem.h"
+#include "ECS/Systems/Implementations/TownSystem.h"
```
Comment (bwrsandman) [inline src/Game.cpp:49]: Why not the interface?

File: src/Renderer.cpp [hunk 14]
```diff
@@ -23,7 +23,7 @@
 #include "3D/Water.h"
 #include "ECS/Components/Sprite.h"
 #include "ECS/Registry.h"
-#include "ECS/Systems/RenderingSystem.h"
+#include "ECS/Systems/Implementations/RenderingSystem.h"
```
Comment (bwrsandman) [inline src/Renderer.cpp:26]: Why not the interface?

**Review by bwrsandman — CHANGES_REQUESTED** [review]

File: src/ECS/Systems/Implementations/DynamicsSystem.cpp [hunk 15]
```diff
@@ -7,6 +7,8 @@
  * openblack is licensed under the GNU General Public License version 3.
  *****************************************************************************/
 
+#define LOCATOR_IMPLEMENTATIONS 1
```
Comment (bwrsandman) [inline src/ECS/Systems/Implementations/DynamicsSystem.cpp:10]: You don't need the 1.
`#define LOCATOR_IMPLEMENTATIONS ` is enough for `#if !defined(LOCATOR_IMPLEMENTATIONS)`

**Review by bwrsandman — CHANGES_REQUESTED** [review]

File: src/ECS/Registry.cpp (hunk 12 above)
Comment (raffclar) [reply src/ECS/Registry.cpp]: Sorted

**Review by bwrsandman — APPROVED** [review]

**Pull request merged**